src/test/run-make-fulldeps/coverage/doctest.rs

```diff
@@ -64,3 +77,23 @@ fn main() {
         assert_eq!(1, 2);
     }
 }
+
+// FIXME(Swatinem): Fix known issue that coverage code region columns need to be offset by the
+// doc comment line prefix (`///` or `//!`) and any additional indent (before or after the doc
+// comment characters). This test produces `llvm-cov show` results demonstrating the problem.
+//
+// One of the above tests now includes: `derive(Debug, PartialEq)`, producing an `llvm-cov show`
+// result with a distinct count for `Debug`, denoted by `^1`, but the caret points to the wrong
+// column. Similarly, the `if` blocks without `else` blocks show `^0`, which should point at, or
+// one character past, the `if` block's closing brace. In both cases, these are most likely off
+// by the number of characters stripped from the beginning of each doc comment line: indent
+// whitespace, if any, doc comment prefix (`//!` in this case) and (I assume) one space character
+// (?). Note, when viewing `llvm-cov show` results in `--color` mode, the column offset errors are
+// more pronounced, and show up in more places, with background color used to show some distinct
+// code regions with different coverage counts.
+//
+// NOTE: Since the doc comment line prefix may vary, one possible solution is to replace each
+// character stripped from the beginning of doc comment lines with a space. This will give coverage
+// results the correct column offsets, and I think it should compile correctly, but I don't know
+// what affect it might have on diagnostic messages from the compiler, and whether anyone would care
+// if the indentation changed. I don't know if there is a more viable solution.
```

Comment:
Thanks for the heads up!

And I thought about doing exactly this, maybe also for the line numbers (just insert blank lines), instead of the offset from the start.

I think it would be a viable hack, but it would mean splitting up the code that generates the "run in the playground" links from the code that actually generates the testcase.
A more general solution would probably be something similar to JS sourcemaps, although they themselves are horrible, lol.

Reply:
I don't know anything about how playground works, but I can't see any reason to run coverage-enabled binaries from playground, so applying this conversion only conditionally (when not executing in playground)--as I think you suggested elsewhere--seems reasonable to me.